pkg/controller/gkenetworkparamset/gkenetworkparamset_controller.go

```diff
@@ -218,6 +218,8 @@ func (c *Controller) reconcile(ctx context.Context, key string) error {
 	originalParams := obj.(*networkv1.GKENetworkParamSet)
 	params := originalParams.DeepCopy()
 
+	gnpObjects.WithLabelValues(originalParams.Status.NetworkName, "", string(originalParams.Spec.DeviceMode)).Dec()
```

Comment:
same comment as in https://github.com/kubernetes/cloud-provider-gcp/pull/534#discussion_r1190742565

Reply:
I agree with Antonio, we should cache the original labels and update metrics right at the end.